pyiron_base/generic/datacontainer.py

```diff
@@ -152,6 +152,18 @@ class DataContainer(MutableMapping):
     [3, 2, 1, 0]
     >>> list(pl.keys())
     [0, 1, 2, 3]
+
+    .. attention:: Subclasses beware!
+
+        DataContainer require some careful treatment when creating subclasses.
+
+        1. Since DataContainers are expected to recursively instantiate themselves subclasses need to keep their
+        `__init__ compatible to the base class.  That means being able to be instantiated without arguments, if
+        arguments are given the first one (or `init`) has to accept a Mapping or Iterable.  Additional arguments may be
+        added, but must be after `init` and must have a default.
+        2. Creating new instance attributes that don't live in the container itself, you need to use
```

Comment:
This one is especially confusing

Reply:
![ralph](https://gfycat.com/emotionalclosedgavial.gif)

I'll re-word it a bit. :/

Reply:
🤣 I just meant the concept, not that the phrasing is so bad. I think this case desperately needs to be shown not just told.